core/src/main/scala/org/http4s/headers/Keep-Alive.scala

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s
+package headers
+
+import cats.implicits._
+import cats.parse._
+import org.http4s.internal.parsing.Rfc7230
+import org.http4s.util.{Renderable, Writer}
+import org.http4s.Header
+import org.typelevel.ci._
+import scala.concurrent.duration.FiniteDuration
+import java.util.concurrent.TimeUnit
+import scala.collection.mutable.ListBuffer
+import cats.kernel.Semigroup
+
+object `Keep-Alive` {
+
+  sealed trait KeepAlive
+  final case class Timeout(timeoutSeconds: Long) extends KeepAlive
+  final case class Max(max: Long) extends KeepAlive
+  final case class Extension(ext: (String, Option[String])) extends KeepAlive
+
+  /*
+https://www.w3.org/Protocols/HTTP/1.1/draft-ietf-http-v11-spec-01#Keep-Alive
+
+Keep-Alive           = "Keep-Alive" ":" 1#keep-alive-info
+keep-alive-info      =   "timeout" "=" delta-seconds
+                       / "max" "=" 1*DIGIT
+                       / keep-alive-extension
+keep-alive-extension = token [ "=" ( token / quoted-string ) ]
+   */
+
+  def apply(
+      timeoutSeconds: Option[Long],
+      max: Option[Long],
+      extension: List[(String, Option[String])]): ParseResult[`Keep-Alive`] =
+    if (timeoutSeconds.isDefined || max.isDefined || extension.nonEmpty) {
+      val validatedTimeoutSeconds = timeoutSeconds.traverse(t => nonNegativeLong(t, "timeout"))
+      val validatedMax = max.traverse(m => nonNegativeLong(m, "max"))
+      (validatedTimeoutSeconds, validatedMax).mapN((t, m) => impl(t, m, extension))
+    } else {
+      ParseResult.fail("Invalid Keep-Alive header", "All fields of Keep-Alive were empty")
+    }
+
+  def unsafeApply(
+      timeoutSeconds: Option[Long],
+      max: Option[Long],
+      extension: List[(String, Option[String])]): `Keep-Alive` =
+    apply(timeoutSeconds, max, extension).fold(throw _, identity)
+
+  def parse(s: String): ParseResult[`Keep-Alive`] =
+    ParseResult.fromParser(parser, "Invalid Keep-Alive header")(s)
+
+  private[http4s] def nonNegativeLong(l: Long, fieldName: String): ParseResult[Long] =
```

Comment:
Why `http4s`?  You can make this private, and we'll get nice compiler warnings if it's ever unused.